hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestMiniDFSCluster.java

```diff
@@ -309,6 +309,14 @@ public void testSetUpFederatedCluster() throws Exception {
             DFSUtil.addKeySuffixes(
             DFS_NAMENODE_HTTP_ADDRESS_KEY, "ns1", "nn1")));
       }
+
+      // Shutdown namenodes individually.
+      cluster.shutdownNameNode(0);
+      cluster.shutdownNameNode(1);
```

Comment:
Shall we just do restarts rather than shutdown + restart?

Reply:
I don't think that will repro the issue, Single restart shall pass only since all other namenodes are up. May be just one shutdown and the other one restarting will repro...
But that won't look good

Reply:
> I don't think that will repro the issue, Single restart shall pass only since all other namenodes are up. May be just one shutdown and the other one restarting will repro...
But that won't look good. My original use case was from a RBF patch, where I was restaring the namenodes like this only, so kept the logic same

Ah I see, yeah makes sense 👍